From 0e23a6368e1f13c2f432856d5018ee30980ba4f9 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:40:13 -0400
Subject: [virt] kvm: return an error code in kvm_vm_ioctl_register_coalesced_mmio()

Message-id: <1406040016-3289-219-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86297
O-Subject: [RHEL7 PATCH v2 218/221] KVM: return an error code in kvm_vm_ioctl_register_coalesced_mmio()
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Dan Carpenter <dan.carpenter@oracle.com>

If kvm_io_bus_register_dev() fails then it returns success but it should
return an error code.

I also did a little cleanup like removing an impossible NULL test.

Cc: stable@vger.kernel.org
Fixes: 2b3c246a682c ('KVM: Make coalesced mmio use a device per zone')
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit aac5c4226e7136c331ed384c25d5560204da10a0)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/virt/kvm/coalesced_mmio.c b/virt/kvm/coalesced_mmio.c
index 88b2fe3..00d8642 100644
--- a/virt/kvm/coalesced_mmio.c
+++ b/virt/kvm/coalesced_mmio.c
@@ -154,17 +154,13 @@ int kvm_vm_ioctl_register_coalesced_mmio(struct kvm *kvm,
  list_add_tail(&dev->list, &kvm->coalesced_zones);
  mutex_unlock(&kvm->slots_lock);
 
- return ret;
+ return 0;
 
 out_free_dev:
  mutex_unlock(&kvm->slots_lock);
-
  kfree(dev);
 
- if (dev == NULL)
-  return -ENXIO;
-
- return 0;
+ return ret;
 }
 
 int kvm_vm_ioctl_unregister_coalesced_mmio(struct kvm *kvm,
-- 
1.7.1